pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	gcpclient "github.com/gardener/gardener-extension-provider-gcp/pkg/internal/client"
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	ctrlerror "github.com/gardener/gardener/extensions/pkg/controller/error"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Bastion, cluster *controller.Cluster) error {
+	logger := a.logger.WithValues("bastion", client.ObjectKeyFromObject(bastion), "operation", "reconcile")
+
+	gcpclient, err := a.getGCPClient(ctx, bastion)
+	if err != nil {
+		return errors.Wrap(err, "failed to create GCP client")
+	}
+
+	opt, err := DetermineOptions(ctx, bastion, cluster)
+	if err != nil {
+		return errors.Wrap(err, "failed to setup GCP options")
+	}
+
+	err = ensureFirewallRule(ctx, bastion, gcpclient, opt)
+	if err != nil {
+		return errors.Wrap(err, "failed to ensure firewall rule")
+	}
+
+	endpoints, err := ensureBastionInstance(ctx, logger, gcpclient, opt)
+	if err != nil {
+		return errors.Wrap(err, "failed to ensure bastion instance")
+	}
+
+	if !endpoints.Ready() {
+		return &ctrlerror.RequeueAfterError{
+			// requeue rather soon, so that the user (most likely gardenctl eventually)
+			// doesn't have to wait too long for the public endpoint to become available
+			RequeueAfter: 5 * time.Second,
+			Cause:        errors.New("bastion instance has no public/private endpoints yet"),
+		}
+	}
+
+	return controller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.Client(), bastion, func() error {
```

Comment:
```suggestion
	// once a public endpoint is available, publish the endpoint on the
	// Bastion resource to notify upstream about the ready instance
	return controller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.Client(), bastion, func() error {
```
add comment from aws provider implementation as it fits well

Reply:
fixed in d5a96db004a9bfcb41aaaa41cc695a0fd062817f